### PR TITLE
consul service discovery sample

### PR DESCRIPTION
**Pull request by zhangshen023**
consul service discovery sample

### CODE REVIEW

File: registry/servicediscovery/consul/go-client/app/client.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+import (
+	hessian "github.com/apache/dubbo-go-hessian2"
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-client/app/client.go:30]: split this import block and move it to the 2rd import block.

File: registry/servicediscovery/consul/go-client/app/client.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+// they are necessary:
+// 		export CONF_CONSUMER_FILE_PATH="xxx"
+// 		export APP_LOG_CONF_FILE="xxx"
+func main() {
+	hessian.RegisterPOJO(&User{})
+	config.Load()
+	time.Sleep(3e9)
+
+	fmt.Printf("\033[32;40m" + "\n\n\nstart to test dubbo" + "\033[0m\n")
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-client/app/client.go:62]: pls using gxlog.CInfo from "github.com/dubbogo/gost/log" instead. as https://github.com/apache/dubbo-go-samples/blob/master/helloworld/dubbo/go-client/app/client.go#L56

File: registry/servicediscovery/consul/go-client/app/client.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	user := &User{}
+	for i := 0; i < 5; i++ {
+		err := userProvider.GetUser(context.TODO(), []interface{}{"A001"}, user)
+		if err != nil {
+			panic(err)
+		}
+		fmt.Printf("\033[32;40m"+"response result: %v\n"+"\033[0m\n", user)
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-client/app/client.go:69]: pls using gxlog.CInfo from "github.com/dubbogo/gost/log" instead. as https://github.com/apache/dubbo-go-samples/blob/master/helloworld/dubbo/go-client/app/client.go#L56

File: registry/servicediscovery/consul/go-server/app/user.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	Name string
+	Age  int32
+	Time time.Time
+}
+
+type UserProvider struct {
+}
+
+func (u *UserProvider) GetUser(ctx context.Context, req []interface{}) (*User, error) {
+	fmt.Printf("\033[32;40m"+"req:%#v"+"\033[0m\n", req)
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-server/app/user.go:48]: pls using gxlog.CInfo from "github.com/dubbogo/gost/log" instead. as https://github.com/apache/dubbo-go-samples/blob/master/helloworld/dubbo/go-client/app/client.go#L56

File: registry/servicediscovery/consul/go-server/app/user.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	rsp := User{"A001", "Alex Stocks", 18, time.Now()}
+	fmt.Printf("\033[32;40m"+"rsp:%#v"+"\033[0m\n", rsp)
```
Comment (AlexStocks) [inline registry/servicediscovery/consul/go-server/app/user.go:50]: pls using gxlog.CInfo from "github.com/dubbogo/gost/log" instead. as https://github.com/apache/dubbo-go-samples/blob/master/helloworld/dubbo/go-client/app/client.go#L56

**Review by AlexStocks — APPROVED** [review]

**Review by zouyx — APPROVED** [review]

**Pull request merged**